src/measure_theory/set_integral.lean

```diff
@@ -0,0 +1,291 @@
+  apply indicator_le_indicator_ae,
+  exact h
+end
+
+lemma integral_on_le_integral_on {f g : α → ℝ} (hfm : measurable_on s f) (hfi : integrable_on s f)
+  (hgm : measurable_on s g) (hgi : integrable_on s g) (h : ∀ a, a ∈ s → f a ≤ g a) :
+  (∫ a in s, f a) ≤ (∫ a in s, g a) :=
+integral_on_le_integral_on_ae hfm hfi hgm hgi $ by filter_upwards [] h
+
+lemma integral_on_union (hsm : measurable_on s f) (hsi : integrable_on s f)
+  (htm : measurable_on t f) (hti : integrable_on t f) (h : disjoint s t) :
+  (∫ a in (s ∪ t), f a) = (∫ a in s, f a) + (∫ a in t, f a) :=
+by { rw [indicator_union_of_disjoint h, integral_add hsm hsi htm hti] }
+
+lemma integral_on_union_ae (hs : is_measurable s) (ht : is_measurable t) (hsm : measurable_on s f)
+  (hsi : integrable_on s f) (htm : measurable_on t f) (hti : integrable_on t f) (h : ∀ₘ a, a ∉ s ∩ t) :
+  (∫ a in (s ∪ t), f a) = (∫ a in s, f a) + (∫ a in t, f a) :=
+begin
+  have := integral_congr_ae _ _ (indicator_union_ae h f),
+  rw [this, integral_add hsm hsi htm hti],
+  { exact hsm.union hs ht htm },
+  { exact hsm.add htm }
+end
+
+lemma tendsto_integral_on_of_monotone {s : ℕ → set α} {f : α → β} (hsm : ∀i, is_measurable (s i))
+  (h_mono : monotone s) (hfm : measurable_on (Union s) f) (hfi : integrable_on (Union s) f) :
+  tendsto (λi, ∫ a in (s i), f a) at_top (nhds (∫ a in (Union s), f a)) :=
+let bound : α → ℝ := indicator (Union s) (λa, ∥f a∥) in
+begin
+  apply tendsto_integral_of_dominated_convergence,
+  { assume i, exact hfm.subset (hsm i) (subset_Union _ _) },
+  { assumption },
+  { show integrable_on (Union s) (λa, ∥f a∥), rwa integrable_on_norm_iff },
+  { assume i, apply all_ae_of_all,
+    assume a,
+    rw [norm_indicator_eq_indicator_norm],
+    exact indicator_le_indicator_of_subset (subset_Union _ _) (λa, norm_nonneg _) _ },
+  { filter_upwards [] λa, le_trans (tendsto_indicator_of_monotone _ h_mono _ _) (pure_le_nhds _) }
+end
+
+lemma tendsto_integral_on_of_antimono (s : ℕ → set α) (f : α → β) (hsm : ∀i, is_measurable (s i))
+  (h_mono : ∀i j, i ≤ j → s j ⊆ s i) (hfm : measurable_on (s 0) f) (hfi : integrable_on (s 0) f) :
+  tendsto (λi, ∫ a in (s i), f a) at_top (nhds (∫ a in (Inter s), f a)) :=
+let bound : α → ℝ := indicator (s 0) (λa, ∥f a∥) in
+begin
+  apply tendsto_integral_of_dominated_convergence,
+  { assume i, refine hfm.subset (hsm i) (h_mono _ _ (zero_le _)) },
+  { exact hfm.subset (is_measurable.Inter hsm) (Inter_subset _ _) },
+  { show integrable_on (s 0) (λa, ∥f a∥), rwa integrable_on_norm_iff },
+  { assume i, apply all_ae_of_all,
+    assume a,
+    rw [norm_indicator_eq_indicator_norm],
+    refine indicator_le_indicator_of_subset (h_mono _ _ (zero_le _)) (λa, norm_nonneg _) _ },
+  { filter_upwards [] λa, le_trans (tendsto_indicator_of_antimono _ h_mono _ _) (pure_le_nhds _) }
+end
+
+-- TODO : prove the following proposition
+
+-- lemma integral_on_Union (s : ℕ → set α) (f : α → β) (hm : ∀i, is_measurable (s i))
```

Comment:
Could you please prove this using #1884?

Reply:
Sorry. I was planning to prove this after the current PR is merged. 

But now it is done.